SETUP.md

```diff
@@ -0,0 +1,300 @@
+
+In the host terminal, open a folder with development projects. Then run:
+```sh
+git clone https://github.com/utat-ss/finch-flight-software.git
+cd finch-flight-software
+```
+
+</details>
+
+
+### 3. Install pyOCD
+
+pyOCD is a tool that facilitates the interaction between the STM32 chip and your computer.
+
+In the host terminal, run:
+
+___________
+
+<details><summary>Windows</summary>
+
+`python -m pip install -U pyocd`
+
+</details>
+
+<details><summary>macOS/Linux</summary>
+
+`python3 -m pip install -U pyocd`
+
+</details>
+
+___________
+
+
+If you have a Raspberry Pi or non-x86 Windows platform to install pyOCD on, check out [this](https://pyocd.io/docs/installing_on_non_x86) pyOCD guide for more information.
+
+### 4. Download and place pyOCD packs
+
+pyOCD needs Keil packs for STM32G4 and STM32H7 to support our OBC and PAY boards. When you run any pyOCD command, you can pass a Keil pack with a `--pack` option. Our scripts also use `--pack` option and expect the packs under a specific path.
+
+Steps:
+
+1. Create folder named `packs` in the `finch-flight-software` folder.
+2. Open [this](https://www.keil.arm.com/packs/stm32h7xx_dfp-keil/overview/) page or search `stm32h7 pack keil`.
+3. Download the .pack file and place it in the `packs` folder.
+4. Rename the .pack file to `stm32h7.pack`.
+5. Open [this](https://www.keil.arm.com/packs/stm32g4xx_dfp-keil/overview/) page or search `stm32g4 pack keil`.
+6. Download the .pack file and place it in the `packs` folder.
+7. Rename the .pack file to `stm32g4.pack`.
+
+
+### 5. Install Docker
+
+Docker is a tool that can run containers (sandboxes) on your machine. We use Docker containers so that everyone has the same development environment.
+
+___________
+
+<details><summary>Windows</summary>
+
+Install Docker Desktop using [this](https://www.docker.com/products/docker-desktop/) link.
+
+After installing, open the Docker Desktop application and keep it running in the background. You can skip the account creation.
+
+</details>
+
+<details><summary>macOS</summary>
+
+For macOS, install [Docker Desktop](https://www.docker.com/products/docker-desktop/).
+
+After installing, open the Docker Desktop application and keep it running in the background. You can skip the account creation.
+
+</details>
+
+<details><summary>Linux</summary>
+
+Please use [this](https://docs.docker.com/engine/install/) guide to install Docker.
+</details>
+
+___________
+
+### 6. Pull the FINCH dev environment image
+
+In the host terminal, run:
+```sh
+docker pull ghcr.io/utat-ss/finch-flight-software:latest
+```
+
+This command pulls the container with our development environment. It might take a few minutes to finish.
+
+## Usage
+
+Our Docker container supports building the code by itself. To flash/debug the board, you need both the host and the container terminal.
+
+The container doesn't have proper access to SSH credentials. For this reason, you need to run `git` commands on the host.
```

Comment:
You mean Git operations that that require interaction with the remote?
Also let's not assume everyone uses SSH (also relates to a comment earlier).